README.md

```diff
@@ -16,3 +16,78 @@ The project and Python dependencies can be installed from source with:
 ```
 pip3 install .
 ```
+
+## Configuration System
+
+This project provides a configuration system that supplies options (e.g., `user`, `token`) to a command-line interface (CLI) tool.
+The configuration system follows a tiered order, incorporating both files and command-line options.
+
+### Configuration Sources
+
+The configuration system loads options from JSON files located across multiple directories.
+It searches for these files when invoked, they are not created automatically.
+By default, it looks for files named `edq-config.json`, this is customizable.
+
+For example, a configuration file containing the `user` and `token` options might look like this:
+```
+{
+    "user": "edq-user",
+    "token": "1234567890"
+}
+```
+
+The table below lists configuration sources in the order they are evaluated, from the first source at the top row to the last source at the bottom row.
+All sources are processed in the order shown in the table: if an option appears only in one source, it is included as is.
+If the same option appears in multiple sources, the value from the later source in the table overrides the earlier one.
```

Comment:
I think we can merge these two sentences.
Take the first part of the first sentence,
and then add on what happens with multiple sources.